### DISCUSSION TITLE
My Production build on google cloud run seems to always include localhost in the callbackUrl

### DISCUSSION THREAD

**Discussion by nobitahuang** [nextauthjs/next-auth#2207] [Help]
## Question 💬

im looking for an answer for in the documentations in the existign issue but seemslike nobody is having the same issue as me

my login url will always be like this
`https://domain.com/sign-in?callbackUrl=http://localhost:3000https://domain.com/`

localhost:3000 is included when the app is on production build

im pretty sure i set the NEXTAUTH_URL to https://domain.com in the env file.

## Contributing 🙌🏽

It takes a lot of work 🏋🏻‍♀️ maintaining a library like `next-auth`; any contribution is more than welcome 💚

In case you're willing to help answer this question, please let us know here, and we'll reach you 😊 . Otherwise, you can have a look at the issues labelled with [`"good first issue"`](https://github.com/nextauthjs/next-auth/issues?q=is%3Aissue+is%3Aopen+label%3A%22good+first+issue%22) and pick any of them.

**Comment by jpomykala** [comment]
I have the same issue

**Comment by allangalera** [comment]
I have the same issue. I'm using gitlab-ci + serverless and I even logged the env to make sure it's right and it is. But still gave me `http://localhost:3000` on providers endpoint.

I didn't have a .env file but I'm setting the env within gitlab so it should be the same. But it's not. I tried to manually create .env file on the pipeline `echo "NEXTAUTH_URL=$NEXTAUTH_URL" > .env` and it worked.

I'm guessing is something with `parseUrl` not receiving the right url so it returns `http://localhost:3000`

**Comment by buehler** [comment]
same issue. it always includes those weird localhost urls for the first signin.
